pkgdb/src/pkgdb/write.cc

```diff
@@ -212,6 +208,36 @@ PkgDb::PkgDb( const nix::flake::LockedFlake & flake, std::string_view dbPath )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+void
+PkgDb::connect()
+{
+  /* The `locking_mode` pragma acquires an exclusive write lock the first time
+   * that the database is written to and only releases the lock once the
+   * database connection is closed. We make a write as soon as possible after
+   * opening the connection to establish the write lock. After the
+   * `RETRY_WHILE_BUSY` call returns we should be the only process able to write
+   * to the database, though other processes mays still read from the database
+   * (this is why we must use `EXCLUSIVE` transactions, which prevent reads as
+   * well).
+   *
+   * It could be the case that this database hasn't been initialized yet, so we
+   * can't write to an existing table. Instead we just write to a dummy table.
+   * It's unclear whether setting a pragma value like `appliation_id` counts as
+   * a write, so we create a table instead.*/
+  static const char * acquire_lock = R"SQL(
```

Comment:
I thought we removed this during pair programming?

Reply:
We did as a means of debugging, but it doesn't work without it.

Reply:
The relevant behavior is documented here: https://www.sqlite.org/pragma.html#pragma_locking_mode

I would recommend this be something like "pragma application_id = 42", not a table creation.